src/resource.ts

```diff
@@ -275,10 +275,14 @@ export class Resource<T = any> extends EventEmitter {
    *
    * This will update the local state but *not* update the server
    */
-  updateCache(state: State<T>) {
+  updateCache(state: State<T>): void {
 
-    this.client.cache.store(state);
-    this.emit('update', state);
+    this.client.cacheState(state);
+
+    const resource = this.client.resources.get(state.uri);
+    if (!resource) {
```

Comment:
I'm not sure why this check is needed. The resource should exist because it should point to `this`.

However, I realize that it's possible to set the cache for a resource with an incorrect uri, so perhaps it's worth checking the `uri` and see if it matches `this.uri` ?

Reply:
I'm not sure if it's needed either, but when reading the code, `cacheState()` would also emit an `update` event, if the resource was in `this.client.resources`, so I added the check to ensure the 'update' event was not triggered twice.